src/pypwsqc/indicator_correlation.py

```
@@ -208,13 +208,22 @@ def ic_filter(
     bins = np.arange(0, max_distance, bin_size)
 
     # quantile parameter not too low, otherwise the line becomes to wiggly - depends on data  # noqa: E501
-    binned_indcorr_ref = (
-        indicator_correlation_matrix_ref.groupby_bins(
-            distance_correlation_matrix_ref, bins=bins
+    try:
+        binned_indcorr_ref = (
+            indicator_correlation_matrix_ref.groupby_bins(
+                distance_correlation_matrix_ref, bins=bins
+            )
+            .quantile(quantile_bin_ref)
+            .bfill(dim="group_bins")
         )
-        .quantile(quantile_bin_ref)
-        .bfill(dim="group_bins")
-    )
+    except ValueError as e:
+        if "([0])" in str(e):
+           print("Insert a valid bin_size")
+           return
```

Should the user be informed here that the ic_filter fails? If yes, then raise again a `ValueError`, but with the meaningful error message that you have in the print statement.

You should not cat an exception and then silently return None because this might results in unexpected behavior of the code that is calling `ic_filter`.

The same applies to the except statements below.